src/pages/blog/starlingx-ptp-redundant-ptp-timing-clock-sources.md

```diff
@@ -0,0 +1,92 @@
+
+And as part of the constant effort to meet the evolving needs of telecommunications industry and to provide a more robust Precision Time Protocol (PTP) deployment, the StarlingX 9.0 introduces the High Availability feature, providing a way to sync multiple Physical Hardware Clock (PHC) in a N+1 redundancy configuration.
+
+The feature allows, in a t-GM application, multiple GPS clock sources connected, and in a t-BC application, multiple PTP sources connected to redundant NICs, provinding protection against failtures.
+
+In the N+1 redundancy configuration, the available clock is selected active and commands the system clock and the downstream PTP clock reference. The remaining clock sources are kept ready to become active when required.
+
+This article will speak about the new HA feature configuration, clock selection algorithm, and new alarms.
+
+If you're interested in learning more about the initial use cases and implementation, you check out an earlier [blog post series](https://www.starlingx.io/blog/starlingx-ptp-part-1/) and ["A Deep Dive Into the Precision Time Protocol Configuration Enhancements"](https://www.starlingx.io/blog/starlingx-ptp-multi-instance-features/).
+
+# Configuration
+
+As pointed out in ["A Deep Dive Into the Precision Time Protocol Configuration Enhancements"](https://www.starlingx.io/blog/starlingx-ptp-multi-instance-features/), when a StarlingX host is configured as a PTP node,  there are two applications used: ptp4l and phc2sys. The ptp4l is responsible for running the Best Master Clock Algorithm (BMCA) and to desciplin the oscilator on the NIC according to the incomming PTP timestamps. And the phc2sys reads the time from a disciplined NIC and synchronizes the system clock with it.
+
+Up to StarlingX version 8.0, the Linux real-time system clock was synchronized to only a single time source (PHC) through static configuration of the phc2sys instance. From release 9.0 and up, when the HA feature enabled, the phc2sys configuration will accept association to multiple ptp4l instances, providing redundancy protection to the clock sources.
+
+When the HA feature enabled, the real-time system clock is synchronized with the best time source available. The operator configure the following clock status values to control the source selection algorithm:
+* GM clock class
+* GM clock accuracy
+* GM offset scaled log variance
+* Local clock class
+* Local clock accuracy
+* Local offset scaled log variance
+* Frequency traceable
+* Time traceable
+
+These parameters set the threshold for a source to be eligible for use in setting the system real-time clock. For example, in a t-GM application, configuring 6 in the maximum local class makes only locked sources eligible for selection.
```

Comment:
Can you please explain this in a bit more detail?

Reply:
This sentence is a bit hard to follow. Does the example of configuring the maximum local class refer to 'GM clock class'?

Reply:
No, it refers to 'Local clock class'. It is a bit confuse to use 'Local clock class' in t-GM configuration, not 'GM clock class'. The clock is local for the system configured as t-GM, because it is directly connected to the GPS clock source.